### DISCUSSION TITLE
Variable interpolation, get the name of a group instead of the configuration provided by that group.

### DISCUSSION THREAD

**Discussion by YerePhy** [facebookresearch/hydra#2250] [Q&A] [answered]
Hi, 

When I try to do varaible interpolation: 

```yaml
defaults:
  - datamodule: monai_holdout
  - lightningmodule: unet_like
  - trainer: standard
  - _self_
datamodule_type: "${datamodule}"
```

Being my main script:

```python
import hydra
from omegaconf import DictConfig, OmegaConf
from utils import datamodule_factory, lightningmodule_factory


@hydra.main(config_path="./configs/train", config_name="config")
def main(cfg: DictConfig):
    """
    Perform training.
    """
    OmegaConf.register_new_resolver("get_method", hydra.utils.get_method)
    cfg = hydra.utils.instantiate(cfg, _recursive_=True)
    
    print(cfg['datamodule_type'])
main()
```

I would like to have in `cfg["datamodule_type"]` the name of the group provided to `datamodule` in the defaults list i.e. `"monai_holdout"`. However, I am getting the configuration provided by that group:

<pre>
Output main script:
{'dataset_path': '/home/yere/Repositories/brain_tumor/datasets/dataset_concat', 'modalities': <enum 'Modalities'>, 'ext': '.nii.gz', 'batch_size': 5, 'num_workers': 1, 'load_subset': 0.01, 'splitter': functools.partial(<function partition_dataset at 0x7fcf1119ca60>, ratios=[0.7, 0.15, 0.15], num_partitions=3), 'patch_func': <monai.transforms.croppad.dictionary.RandCropByLabelClassesd object at 0x7fcef388e730>, 'train_stage_transformations': <monai.transforms.compose.Compose object at 0x7fcef3868a30>, 'test_stage_transformations': <monai.transforms.compose.Compose object at 0x7fcef389c940>}
</pre>

Is it possible to have only the name of the group by variable interpolation or by other method?

**Comment by Jasha10** [comment, marked as answer]
I believe this is currently not possible. Once Hydra composes the config object, the defaults lists are erased and the information about "where a given subconfig came from" is gone.
One idea I have seen before is to add `name: monai_holdout` to your `monai_holdout.yaml` file.

**Reply by YerePhy** [reply]
Okay, thanks for pointing that out @Jasha10, I was doing something similar but now I've changed to this approach.
